File: android/app/src/main/kotlin/com/example/figure_skating_jumps/x_sens_dot/enums/RecordingStatus.kt
package com.example.figure_skating_jumps.x_sens_dot.enums

/**
 * A enum class that contains the XSens Dot recording status
 *
 * @param status The recording status
 */
enum class RecordingStatus(val status: String) {
    SetRate("SetRate"),
    EnableRecordingNotificationDone("EnableRecordingNotificationDone"),
    RecordingStarted("RecordingStarted"),
    RecordingStopped("RecordingStopped"),
    GetFlashInfoDone("GotFlashInfo"),
    GetFileInfoDone("GotFileInfo"),
    ExtractingFile("ExtractingFile"),
    ExtractFileDone("ExtractFileDone"),
    EraseMemoryDone("EraseMemoryDone")
}

File: lib/enums/recording/recording_status.dart
enum RecordingStatus {
  setRate("SetRate"),
  enableRecordingNotificationDone("EnableRecordingNotificationDone"),
  recordingStarted("RecordingStarted"),
  recordingStopped("RecordingStopped"),
  getFlashInfoDone("GotFlashInfo"),
  getFileInfoDone("GotFileInfo"),
  extractingFile("ExtractingFile"),
  extractFileDone("ExtractFileDone");

  const RecordingStatus(this.status);

  final String status;
}
